src/QirTools/pyqir/pyqir/__init__.py

```diff
@@ -1,59 +1,214 @@
+# Copyright(c) Microsoft Corporation.
+# Licensed under the MIT License.
+
 from .pyqir import *
 
+
 class QirBuilder:
+    """
+    The QirBuilder object describes quantum circuits and emits QIR
+
+    :param module: name of the QIR module
+    :type module: str
+    """
+
     def __init__(self, module: str):
         self.pyqir = PyQIR(module)
 
-    def cx(self, control:str, target:str):
+    def cx(self, control: str, target: str):
+        """
+        Applies controlled X operation to the target qubit
+
+        :param control: name of the control qubit
+        :type control: str
+        :param target: name of the target qubit
+        :type target: str
+        """
         self.pyqir.cx(control, target)
 
-    def cz(self, control:str, target:str):
+    def cz(self, control: str, target: str):
+        """
+        Applies controlled Z operation to the target qubit
+
+        :param control: name of the control qubit
+        :type control: str
+        :param target: name of the target qubit
+        :type target: str
+        """
         self.pyqir.cz(control, target)
 
     def h(self, target: str):
+        """
+        Applies H operation to the target qubit
+
+        :param target: name of the target qubit
+        :type target: str
+        """
         self.pyqir.h(target)
 
     def m(self, qubit: str, target: str):
+        """
+        Applies measurement operation or the source qubit into the target register
+
+        :param qubit: name of the source qubit
+        :type qubit: str
+        :param target: name of the target register
+        :type target: str
+        """
         self.pyqir.m(qubit, target)
 
     def reset(self, target: str):
+        """
+        Applies Reset operation to the target qubit
+
+        :param target: name of the target qubit
+        :type target: str
+        """
         self.pyqir.reset(target)
 
     def rx(self, theta: float, qubit: str):
+        """
+        Applies Rx operation to the target qubit
```

Comment:
Similarly here, there's a lot of different conventions for rotation operations such that either explicitly listing the matrix form of the unitary that can be used to simulate this operation or linking to Base Profile docs would be really helpful in disambiguiating.